commit 231509953cc11442d15220b37f13e1a1256edb53
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:14 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-macports-stdlibc-only-on-darwin.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index daa379ab6..25877a353 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -163,10 +163,10 @@ proc portconfigure::stdlib_trace {opt action args} {
 
 # helper function to set configure.cxx_stdlib
 proc portconfigure::configure_get_cxx_stdlib {} {
-    global cxx_stdlib compiler.cxx_standard
+    global cxx_stdlib compiler.cxx_standard os.platform
     if {${compiler.cxx_standard} eq ""} {
         return ""
-    } elseif {${cxx_stdlib} eq "libstdc++" && ${compiler.cxx_standard} >= 2011} {
+    } elseif {${os.platform} eq "darwin" && ${cxx_stdlib} eq "libstdc++" && ${compiler.cxx_standard} >= 2011} {
         return macports-libstdc++
     } else {
         return ${cxx_stdlib}
